lib/react_client.dart

```diff
@@ -511,12 +511,13 @@ class JsComponent2Adapter extends Component2Adapter {
   @override
   void setStateWithUpdater(StateUpdaterCallback stateUpdater, SetStateCallback callback) {
     final firstArg = allowInterop((jsPrevState, jsProps, [_]) {
-      return jsBackingMapOrJsCopy(stateUpdater(
+      dynamic value = stateUpdater(
         new JsBackedMap.backedBy(jsPrevState),
         new JsBackedMap.backedBy(jsProps),
-      ));
+      );
+      if (value == null) return null;
```

Comment:
instead of this we could do:
```
      if (value is Map) return jsBackingMapOrJsCopy(value);
      return value;
```

thoughts? @greglittlefield-wf @aaronlademann-wf @joebingham-wk @sydneyjodon-wk

Reply:
StateUpdaterCallback has a return type Map
`typedef Map StateUpdaterCallback(Map prevState, Map props);`

Reply:
I feel like the null check is a little cleaner - it's more common practice (from what I've seen) and easier to reason about.